lib/web_ui/lib/src/engine/platform_dispatcher/view_focus_binding.dart

```diff
@@ -85,18 +75,39 @@ final class ViewFocusBinding {
         direction: _viewFocusDirection,
       );
     }
+    _markViewAsFocusable(_lastViewId, reachableByKeyboard: true);
+    _markViewAsFocusable(viewId, reachableByKeyboard: false);
     _lastViewId = viewId;
-    _notify(event);
+    _onViewFocusChange(event);
   }
 
-  static int? _viewId(DomElement? element) {
-    final DomElement? viewElement = element?.closest(
-      DomManager.flutterViewTagName,
-    );
-    final String? viewIdAttribute = viewElement?.getAttribute(
-      GlobalHtmlAttributes.flutterViewIdAttributeName,
-    );
-    return viewIdAttribute == null ? null : int.tryParse(viewIdAttribute);
+  int? _viewId(DomElement? element) {
+    final DomElement? rootElement = element?.closest(DomManager.flutterViewTagName);
+    if (rootElement == null) {
+      return null;
+    }
+    return _viewManager.viewIdForRootElement(rootElement);
+  }
+
+  void _handleViewCreated(int viewId) {
+    _markViewAsFocusable(viewId, reachableByKeyboard: true);
```

Comment:
If we were to listen to focusin/out, we'd need to use this method to attach listener to new views.